pkg/engine/engine.go

```diff
@@ -492,10 +584,144 @@ func (e *Engine) detectorWorker(ctx context.Context) {
 		}
 		atomic.AddUint64(&e.metrics.ChunksScanned, 1)
 	}
+
+	wgVerificationOverlap.Wait()
 	wgDetect.Wait()
 	ctx.Logger().V(4).Info("finished scanning chunks")
 }
 
+// chunkSecretKey ties secrets to the specific detector that found them. This allows identifying identical
+// credentials extracted by multiple different detectors processing the same chunk. Or duplicates found
+// by the same detector in the chunk. Exact matches on lookup indicate a duplicate secret for a detector
+// in that chunk - which is expected and not malicious. Those intra-detector dupes are still verified.
+type chunkSecretKey struct {
+	secret       string
+	detectorInfo ahocorasick.DetectorInfo
+}
+
+func likelyDuplicate(ctx context.Context, val chunkSecretKey, dupes map[chunkSecretKey]struct{}) bool {
+	const similarityThreshold = 0.9
+
+	valStr := val.secret
+	for dupeKey := range dupes {
+		dupe := dupeKey.secret
+		// Avoid comparing strings of vastly different lengths.
+		if len(dupe)*10 < len(valStr)*9 || len(dupe)*10 > len(valStr)*11 {
+			continue
+		}
+
+		// If the detector type is the same, we don't need to compare the strings.
+		// These are not duplicates, and should be verified.
+		if val.detectorInfo.Type() == dupeKey.detectorInfo.Type() {
+			continue
+		}
+
+		if valStr == dupe {
+			ctx.Logger().V(2).Info(
+				"found exact duplicate",
+			)
+			return true
+		}
+
+		similarity := strutil.Similarity(valStr, dupe, metrics.NewLevenshtein())
+
+		// close enough
+		if similarity > similarityThreshold {
+			ctx.Logger().V(2).Info(
+				"found similar duplicate",
+			)
+			return true
+		}
```

Comment:
I think the last consideration I have for this PR is that we need to let the user know that we've encountered the same/similar finding across multiple different detectors and will not be verifying the result for their safety, and they can override the behavior with --allow-verification-overlap. The current messages are not adequate. It's probably best surfaced through VerificationError, where it will show in the output.

Reply:
This could always be release in audit/warn mode to gauge impact before enabling it in a later release.

Reply:
Added verification erro